src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -310,6 +316,30 @@ protected Task OnPingMessageAsync(PingMessage pingMessage)
             return _serviceMessageHandler.HandlePingAsync(pingMessage);
         }
 
+        private Task OnClientInvocationAsync(ClientInvocationMessage message)
+        {
+            ClientInvocationManager.Router.AddInvocation(message.ConnectionId, message.InvocationId, message.CallerServerId, default);
```

Comment:
Can we link to `ClientConnectionContext.LifetimeTask` instead of this __default__ `CancellationToken`? So we don't need to call `CleanupInvocationsByConnection()` when client connection is aborted.

Reply:
Unimplementable. `CleanupInvocationsByConnection()` is called in `RemoveClientConnection`